todo-app/src/components/NavbarComponent.vue

```diff
@@ -0,0 +1,99 @@
+<template>
+    <nav>
+        <v-app-bar flat app>
+            <v-app-bar-nav-icon
+                class="grey--text"
+                v-if="!appConfig.permanent"
+                @click.stop="appConfig.drawer = !appConfig.drawer"
+            ></v-app-bar-nav-icon>
+            <v-toolbar-title class="text-uppercase grey--text display-1">
+                <span class="font-weight-light">ToDo</span>
+                <span>App</span>
+            </v-toolbar-title>
+
+            <v-spacer></v-spacer>
+
+            <v-btn color="red lighten-1" @click="signOut">
+                <span>Sign Out</span>
+                <v-icon right>mdi-exit-to-app</v-icon>
+            </v-btn>
+        </v-app-bar>
+
+        <v-navigation-drawer
+            v-model="appConfig.drawer"
+            app
+            absolute
+            :temporary="appConfig.overlay && !appConfig.permanent"
+            :permanent="appConfig.permanent"
+            class="gray lighten-2"
+        >
+            <v-list-item class="py-1">
+                <v-list-item-avatar>
+                    <v-img :src="profile.image"></v-img>
+                </v-list-item-avatar>
+
+                <v-list-item-title>{{ profile.username }}</v-list-item-title>
+            </v-list-item>
+
+            <v-divider></v-divider>
+
+            <v-list rounded>
+                <v-list-item
+                    v-for="item in navItems"
+                    :key="item.title"
+                    router
+                    :to="item.route"
+                >
+                    <v-list-item-icon>
+                        <v-icon>{{ item.icon }}</v-icon>
+                    </v-list-item-icon>
+
+                    <v-list-item-content>
+                        <v-list-item-title>{{ item.title }}</v-list-item-title>
+                    </v-list-item-content>
+                </v-list-item>
+            </v-list>
+        </v-navigation-drawer>
+    </nav>
+</template>
+
+<script>
+export default {
+    data() {
+        return {
+            profile: {
+                username: this.$store.getters.profile.username,
+                image: this.$store.getters.profile.img,
+            },
+            appConfig: {
```

Comment:
why is this in app configs 
this is related to drawer so can we think of better names here?

Reply:
I have changed it to navConfig.